src/DataObject/DataObjectDocument.php

```diff
@@ -390,7 +412,21 @@ public function getDependentDocuments(): array
         $dataObjectClasses = array_filter($searchableClasses, function ($class) {
             return is_subclass_of($class, DataObject::class);
         });
-        $ownedDataObject = $this->getDataObject();
+
+        try {
+            $ownedDataObject = $this->getDataObject();
+        } catch (Throwable $e) {
+            // If there is no data object available, it might have been a deleted non-versioned object.
+            // In this case, return empty dependency documents with the assumption that any dependencies
+            // will be handled separately.
+            if (!DataObject::has_extension($this->getSourceClass(), Versioned::class)) {
```

Comment:
can we log before returning just in case we swallow a real error and it becomes very hard to track down